Add localization to data table views

wilkie opened the pull request:
**What**: Localizes the raw strings within the data table.

**Why**: To allow other locales and translations for the data table.

**Note**: Also localizes "Clear Table" since it was related, even though it is not explicitly mentioned in the ticket. Also "Actions" and the various import/export, confirmation dialogs, etc... some of which already have localization.

Very simple changes to add localization keys for the strings used in the table into the common i18n JavaScript file. Though, dozens of simple changes!! Sorry for that!!

Reusing localization for: `Add`, `Cancel`, `Yes`, `Delete`, `Deleting...`, `Clear Table`, 'Edit', 'Saving...', 'Save', `Actions`, and some confirmation text

Adding localization for: `addingToTable`, `addPairToTable`, `addRowToTable`, `backToData`, `clearTable`, `confirmImportOverwriteTitle`, `dataTableConvertToString`, `dataTableConvertToNumber`, `dataTableConvertToBoolean`, `dataTableDebugView`, `dataTableTableView`, `dataTableKey`, `dataTableValue`, `deleteTable`, `deleteTableConfirm`, `enterText`, `exportToCSV`, and `importCSV`

<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->

## Links

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()
- jira ticket: []()
-->

- jira ticket: [FND-2070](https://codedotorg.atlassian.net/jira/software/c/projects/FND/boards/62?modal=detail&selectedIssue=FND-2070)

## Testing story

Manual testing to see the localization strings being used on a development server.

This view now has localized strings for "Add" and "Delete" which already exist! (reuses `add`, `delete`, and `actions`)

![image](https://user-images.githubusercontent.com/31674/186796164-bab74143-87b3-4272-ab0f-42748c74d705.png)

The delete confirmation dialog now has localized strings: (adds `deleteTable` and `deleteTableConfirm`, reuses `delete`, `cancel`)

![image](https://user-images.githubusercontent.com/31674/186796304-d75286fe-3146-4ed6-b102-e965ea0ae550.png)

The data table view is localized: (adds `clearTable`, `importCSV`, `exportToCSV`, `addRowToTable`, `backToData`, `dataTableDebugView`, `dataTableTableView`)

![image](https://user-images.githubusercontent.com/31674/186927113-f3899192-0010-4a95-b6fb-03e71c7644f0.png)

The pending text is localized when adding a row: (adds `addingToTable` to match name convention with `addingToProjects`)

![image](https://user-images.githubusercontent.com/31674/186798334-884a7d49-16e8-4938-a654-2fd1de1be6e1.png)

The column edit dropdown is localized: (adds `convertToString`, `convertToNumber`, `convertToBoolean`; reuses `delete` and `rename`)

![image](https://user-images.githubusercontent.com/31674/186797456-2f9f229b-cadd-471d-a745-a216db8e6d7a.png)

The table editor rows are all localized: (reuses `edit`, `delete`, `deletingWithEllipsis`, `save`)

![image](https://user-images.githubusercontent.com/31674/186798980-6cd5bbe2-859d-4f88-a96c-5197bd24bbc4.png)

The confirm import dialog is localized (existing localization, somehow):

![image](https://user-images.githubusercontent.com/31674/186927508-c5d287e7-05ec-4abb-a3cf-9efef0a0519c.png)

The key-value pairs table view is localized: (reuses `edit`, `save`, `delete`, `saving`)

![image](https://user-images.githubusercontent.com/31674/186935175-10d2530f-d107-4796-9dca-2278e29b68dc.png)

Automated testing to ensure that the strings are passed to rendered objects within the `AddKeyRow`, `AddTableRow`, `AddTableListRow`, `ColumnMenu`, `ConfirmDeleteButton`, `ConfirmImportButton`, `DataTable`, `DataTableView`, `EditKeyRow`, `EditTableListRow`, `EditTableRow`, `KVPairs`, and `TableControls` elements.

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

daynew commented on apps/src/storage/dataBrowser/DataTableView.jsx:
```diff
@@ -122,7 +123,7 @@ class DataTableView extends React.Component {
             >
               <FontAwesome icon="arrow-circle-left" />
-              &nbsp;Back to data
+              &nbsp;{msg.backToData()}
```
Could you do quick manual check that this looks good in RTL languages? I'm worried the spacing might be off. 

Also, does the 'arrow-circle-left' look correct as well?

wilkie replied:
I believe I have resolved this.

daynew reviewed (approved):
Looks good. Great tests!
Just one minor concern

smusoke reviewed (approved):
LGTM

wilkie commented:
Fixing up styling for right-to-left languages and discovered that the data library is hidden from view:

![image](https://user-images.githubusercontent.com/31674/196065308-ed09ef1a-20fa-4f0f-957b-967c36880fc6.png)

Therefore, I have added the correct style to fix this. (adding a `left: 0` to the library pane regardless of direction... it doesn't need to be on one side or the other, logically)

But, either way, I have added the appropriate styles for the existing view:

![image](https://user-images.githubusercontent.com/31674/196065339-b718c60d-2109-4586-b896-8671e7f73573.png)

![image](https://user-images.githubusercontent.com/31674/196065352-655f64b5-f17a-4564-b94e-2ffd6ba86e8a.png)

**NOTE**: The markdown for the Arabic localization seems to be messed up with an extra space, so the links are very broken.

**NOTE**: The right-to-left language support (and language localization support in general) for the library module listing is dubious. It constructs '5 tables', etc, programmably and this is likely very bogus for most languages, but especially right-to-left ones as it renders horribly (But I'm not fixing that, here... we should localize this table separately as it appears in the levelbuilder as well):

![image](https://user-images.githubusercontent.com/31674/196066062-9f9807e5-ff51-4079-b061-c1a9fa035c8b.png)

smusoke commented on apps/src/storage/dataBrowser/DataTableView.jsx:
```diff
@@ -22,6 +23,7 @@ const INITIAL_STATE = {
 class DataTableView extends React.Component {
   static propTypes = {
     // from redux state
+    isRtl: PropTypes.bool.isRequired,
```
Out of curiosity is the `isRtl` pattern an existing one we already implement elsewhere to handle RTL languages?

daynew replied:
yeah, it's all over the place unfortunately.

smusoke commented on apps/test/unit/storage/dataBrowser/AddTableListRowTest.jsx:
```diff
@@ -0,0 +1,37 @@
+
+import AddTableListRow from '@cdo/apps/storage/dataBrowser/AddTableListRow';
+
+describe('AddTableListRow', () => {
```
✨ ✨ Thanks for all of the new tests ✨ ✨
